src/daemon/metric.h

```diff
@@ -32,21 +32,24 @@
 #include "utils/strbuf/strbuf.h"
 #include "utils_time.h"
 
-#define VALUE_TYPE_GAUGE 1
-#define VALUE_TYPE_DERIVE 2
+#define METRIC_ATTR_DOUBLE 0x01
+#define METRIC_ATTR_CUMULATIVE 0x02
 
 typedef enum {
-  METRIC_TYPE_COUNTER = 0,
-  METRIC_TYPE_GAUGE = 1,
-  METRIC_TYPE_UNTYPED = 2,
+  METRIC_TYPE_UNTYPED = 0,
+  METRIC_TYPE_COUNTER = METRIC_ATTR_CUMULATIVE,
+  METRIC_TYPE_FPCOUNTER = METRIC_ATTR_DOUBLE | METRIC_ATTR_CUMULATIVE,
+  METRIC_TYPE_GAUGE = METRIC_ATTR_DOUBLE,
```

Comment:
IMHO clearer to define them in numerical order:
```suggestion
  METRIC_TYPE_UNTYPED = 0,
  METRIC_TYPE_GAUGE = METRIC_ATTR_DOUBLE,
  METRIC_TYPE_COUNTER = METRIC_ATTR_CUMULATIVE,
  METRIC_TYPE_FPCOUNTER = METRIC_ATTR_DOUBLE | METRIC_ATTR_CUMULATIVE,
```

Reply:
🤷  I was going for alphabetic order, but really don't care. Changed.